File: app/javascript/tailmix/runtime/operations/comparison.js
// Operations related to comparison.
export const ComparisonOperations = {
    eq: async (interpreter, args, context) => {
        const val1 = await interpreter.eval(args[0], context);
        const val2 = await interpreter.eval(args[1], context);
        return val1 === val2;
    },

    lt: async (interpreter, args, context) => {
        const val1 = await interpreter.eval(args[0], context);
        const val2 = await interpreter.eval(args[1], context);
        return val1 < val2;
    },

    gt: async (interpreter, args, context) => {
        const val1 = await interpreter.eval(args[0], context);
        const val2 = await interpreter.eval(args[1], context);
        return val1 > val2;
    },
};

File: lib/tailmix/definition/scripting/operations/comparison.rb
# frozen_string_literal: true

module Tailmix
  module Definition
    module Scripting
      module Operations
        # Contains comparison operations.
        module Comparison
          extend self

          def eq(interpreter, args)
            interpreter.eval(args[0]) == interpreter.eval(args[1])
          end

          # def not_eq(interpreter, args)
          #   interpreter.eval(args[0]) != interpreter.eval(args[1])
          # end

          def lt(interpreter, args)
            interpreter.eval(args[0]) < interpreter.eval(args[1])
          end

          def gt(interpreter, args)
            interpreter.eval(args[0]) > interpreter.eval(args[1])
          end
        end
      end
    end
  end
end
